packages/wasm/src/view-server.ts

```diff
@@ -73,22 +80,13 @@ export class ViewServer implements ViewServerInterface {
   // As blocks are scanned, the internal wasmViewServer tree is being updated.
   // Flush updates clears the state and returns all the updates since the last checkpoint.
   flushUpdates(): ScanBlockResult {
-    const raw = this.wasmViewServer.flush_updates() as JsonValue;
-    const { height, sct_updates, new_notes, new_swaps } = raw as {
-      height?: string | number | bigint | undefined;
-      sct_updates?: JsonObject | undefined;
-      new_notes?: JsonValue[] | undefined;
-      new_swaps?: JsonValue[] | undefined;
-    };
-    const wasmJson = {
-      new_notes: (new_notes ?? []).map(n => JSON.stringify(n)),
-      new_swaps: (new_swaps ?? []).map(s => JSON.stringify(s)),
-    };
+    const result = this.wasmViewServer.flush_updates() as FlushResult;
+    const { height, sct_updates, new_notes, new_swaps } = result;
     return {
-      height: BigInt(height ?? false),
+      height: BigInt(height ?? 0),
       sctUpdates: validateSchema(SctUpdatesSchema, sct_updates),
```

Comment:
is it possible to avoid this schema and emit a type for sctUpdates from wasm, or use a pb type

Reply:
It would be nice, but there aren't protobufs for any of these. Added an issue: https://github.com/penumbra-zone/penumbra/issues/4118